lib/bolt/module_installer/specs/forge_spec.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require 'semantic_puppet'
+require 'set'
+
+require 'bolt/error'
+
+# This class represents a Forge module specification.
+#
+module Bolt
+  class ModuleInstaller
+    class Specs
+      class ForgeSpec
+        NAME_REGEX    = %r{\A[a-z][a-z0-9_]*[-/](?<name>[a-z][a-z0-9_]*)\z}.freeze
+        REQUIRED_KEYS = Set.new(%w[name]).freeze
+        KNOWN_KEYS    = Set.new(%w[name version_requirement]).freeze
+
+        attr_reader :full_name, :name, :semantic_version, :type
+
+        def initialize(name, version_requirement = nil)
+          @full_name, @name                       = parse_name(name)
+          @version_requirement, @semantic_version = parse_version_requirement(version_requirement)
+          @type                                   = :forge
+        end
+
+        def self.from_hash(hash)
+          new(hash['name'], hash['version_requirement'])
```

Comment:
So I believe `Target` datatype has this function so it can be called directly from plans. Since this isn't a public API and this function isn't doing any validation or data munging it seems simpler to call `.new()` directly.